### DISCUSSION TITLE
Volatility stop problem

### DISCUSSION THREAD

**Discussion by aiv1** [DaveSkender/Stock.Indicators#1148] [Help and support]
Hi!
I get data from IBKR tws api and want to calculate the volatility stop. Results are strange. See last column:
```
      dateTime      open      high       low     close   volume        rsi        volStp
0   2022-12-19  11656.25  11656.25  11656.25  11656.25        0        NaN           NaN
1   2022-12-20  11639.00  11639.00  11639.00  11639.00        0        NaN           NaN
2   2022-12-21  11802.00  11802.00  11802.00  11802.00        0        NaN           NaN
3   2022-12-22  11512.00  11512.00  11512.00  11512.00        0        NaN           NaN
4   2022-12-23  11534.00  11534.00  11534.00  11534.00        0        NaN           NaN
..         ...       ...       ...       ...       ...      ...        ...           ...
266 2024-01-16  16916.50  17034.00  16855.00  16978.50  1026057  61.200523  2.942384e+06
267 2024-01-17  16825.00  16885.25  16689.00  16859.50   909154  55.975640  2.543360e+06
268 2024-01-18  16996.75  17126.00  16942.50  17117.00  1044730  63.280798  2.852702e+06
269 2024-01-19  17190.50  17472.00  17166.25  17463.25   981099  70.394639  2.466864e+06
270 2024-01-22  17528.25  17584.75  17434.25  17455.50   884567  70.067432  2.797342e+06
```
My code:
```
        df = pandas.DataFrame(self.data, columns=['dateTime', 'open', 'high', 'low', 'close', 'volume'])
        df['dateTime'] = pandas.to_datetime(df['dateTime'], format="%Y%m%d") 
        df.to_csv('MNQ_Hourly.csv')  

        df['rsi'] = talib.RSI(df['close'])

        quotes_list = [
            Quote(d,o,h,l,c,v) 
            for d,o,h,l,c,v 
            in zip(df['dateTime'], df['open'], df['high'], df['low'], df['close'], df['volume'] )
        ]
        results = indicators.get_volatility_stop(quotes_list)
        results_list = [r.sar for r in results]
        df['volStp'] = results_list
        print(df)
```

**Comment by DaveSkender** [comment]
For volatility stop, and many other indicators, they rely on an initialization window that determines how they'll start calculating.  As a result, the [response values](https://dotnet.stockindicators.dev/indicators/VolatilityStop/#response) for those incalculable periods will be `None` or `NaN` in your case.  For this indicator specifically, that warmup period can vary, depending on how long it takes for an identifiable trend reversal.

> The first trend will have `None` values since it is not accurate and based on an initial guess.

**Reply by DaveSkender** [reply]
As a side note, Volatility Stop doesn't use `volume`, so it's not an issue here, but it's not clear why your those values are `0` in your depiction.

```bash
      dateTime      open      high       low     close   volume        rsi        volStp
0   2022-12-19  11656.25  11656.25  11656.25  11656.25        0        NaN           NaN
...
```

**Reply by DaveSkender** [reply]
@LeeDongGeon1996 do you expect `NaN` or `None` here?  I'm unclear on how `null` resolves in our Python implementation.

**Reply by aiv1** [reply]
Hi Dave, my problem are latest values like 2.942384e+06. I would expect something closer to the close price.

**Reply by DaveSkender** [reply]
Oh, I see, those do look unusual.  What package version are you using?

**Reply by DaveSkender** [reply]
Use explicit argument values, 

`indicators.get_volatility_stop(quotes_list, 7, 3)`

maybe our default values aren’t setting correctly.

**Reply by aiv1** [reply]
I've tried 1.1.0 and 1.2.0, same output

**Reply by aiv1** [reply]
same result with indicators.get_volatility_stop(quotes_list, 7, 3)

**Reply by DaveSkender** [reply]
Our unit tests are running okay.  Are `upper_band` and `lower_band` similarly damaged?

**Reply by aiv1** [reply]
yes, same issue in bands

**Reply by LeeDongGeon1996** [reply]
@DaveSkender So, it's not the problem about `None`? 
I'm wondering how the results will be like in the C# package.

**Reply by DaveSkender** [reply]
Looks fine in the C# package, v2.5.0 (same underlying engine in v1.2.0 of the Python package).
https://dotnetfiddle.net/ImEkl4

@aiv1 upload your raw quotes and I'll try to run them against the C# library.  I'll also be able to see if there's something goofy happening with the source quote data that might be throwing off the results.

Any chance there's something happening in your DataFrames syntax?
